packages/origin-ui-irec-core/src/components/detailView/DeviceDetailView.tsx

```diff
@@ -0,0 +1,309 @@
+
+interface IProps {
+    id?: string;
+    externalId?: IExternalDeviceId;
+    showSmartMeterReadings: boolean;
+    showCertificates: boolean;
+}
+
+export function DeviceDetailView(props: IProps) {
+    // should be removed
+    const configuration = useSelector(getConfiguration);
+    const allDevices = useSelector(getAllDevices);
+    const backendClient = useSelector(getBackendClient);
+    // const certificatesClient = useSelector(getCertificatesClient);
+    const organizationClient = backendClient?.organizationClient;
+
+    const originContext = useOriginConfiguration();
+    const originBgColor = originContext?.styleConfig?.MAIN_BACKGROUND_COLOR;
+    const originTextColor = originContext?.styleConfig?.TEXT_COLOR_DEFAULT;
+    const originSimpleTextColor = originContext?.styleConfig?.SIMPLE_TEXT_COLOR;
+
+    const { t } = useTranslation();
+
+    const bgColorDarken = LightenColor(originBgColor, -2);
+    const attributionTextColor = LightenColor(originBgColor, 20);
+    const textColorDarken = LightenColor(originTextColor, -4);
+    const textColorLighten = LightenColor(originTextColor, 52);
+    const bgColorLighten = LightenColor(originBgColor, 5);
+
+    const useStyles = makeStyles(() =>
+        createStyles({
+            attributionText: {
+                fontSize: '10px',
+                color: attributionTextColor
+            }
+        })
+    );
+
+    const classes = useStyles(useTheme());
+
+    const [selectedDevice, setSelectedDevice] = useState<ComposedPublicDevice>(null);
+    useEffect(() => {
+        if (props.id !== null && props.id !== undefined) {
+            const device = allDevices.find((d) => d.id === props.id);
+            setSelectedDevice(device);
+        } else if (props.externalId) {
+            const device = allDevices.find((p) =>
+                p.externalDeviceIds?.find(
+                    (deviceExternalId) =>
+                        deviceExternalId.id === props.externalId.id &&
+                        deviceExternalId.type === props.externalId.type
+                )
+            );
+            setSelectedDevice(device);
+        }
+    }, [props.id, props.externalId]);
+
+    const [tooltip, setTooltip] = useState<string>('');
+    const [deviceTypeImage, setDeviceTypeImage] = useState();
+    useEffect(() => {
+        let isMounted = true;
+        if (isMounted) {
+            const [newTooltip, newImage] = selectIconOnDeviceType(selectedDevice, t);
+            setTooltip(newTooltip);
+            setDeviceTypeImage(newImage);
+        }
+        return () => {
+            isMounted = false;
+        };
+    }, [selectedDevice]);
+
+    const [meterReadCertified, setMeterReadCertified] = useState(null);
+    useEffect(() => {
+        const getMeterRead = async (id: string): Promise<void> => {
+            if (id) {
+                const meterReads = '0';
+                // expects 4 args: id, start??, end??, blockchain account
```

Comment:
Let's use the current year. The requirement for a blockchain account is a mistake, should not be there.